src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs

```diff
@@ -65,10 +65,12 @@ protected AbstractBuiltInCodeStyleDiagnosticAnalyzer(ImmutableArray<DiagnosticDe
                     customTags: DiagnosticCustomTags.Create(isUnnecessary, isConfigurable, enforceOnBuild));
 #pragma warning restore RS0030 // Do not used banned APIs
 
+        // Code style analyzers should not run on generated code, unless explicitly required by the analyzer.
```

Comment:
Might be better as a doc comment. Probably also extend the statement to say it should either be None or Analyze, but not ReportDiagnostic?

Reply:
> Probably also extend the statement to say it should either be None or Analyze, but not ReportDiagnostic?

Hmm, I did not want to be so pedantic given we may have some analyzer that also wants to report on generated code? I can actually make this simple by just making this a bool property though, which may make the purpose more explicit.